pkg/util/log/crash_reporting.go

```diff
@@ -141,12 +134,36 @@ type WrappedPanic struct {
 	Err       interface{}
 }
 
-func sendCrashReport(ctx context.Context, r interface{}) {
-	if err, ok := r.(error); ok {
-		id := raven.CaptureErrorAndWait(err, nil)
-		Shout(ctx, Severity_ERROR, "Reported as error "+id)
+type stringErr string
+
+func (s stringErr) Error() string {
+	return string(s)
+}
+
+var _ error = stringErr("")
+
+var crdbPaths = []string{"github.com/cockroachdb/cockroach"}
+
+func sendCrashReport(ctx context.Context, r interface{}, depth int) {
+	if !DiagnosticsReportingEnabled.Get() || !crashReports.Get() {
+		return // disabled via settings.
+	}
+	if raven.DefaultClient == nil {
+		return // disabled via empty URL env var.
+	}
+
+	var err error
+	if e, ok := r.(error); ok {
+		err = e
 	} else {
-		id := raven.CaptureMessageAndWait(fmt.Sprint(r), nil)
-		Shout(ctx, Severity_ERROR, "Reported as error "+id)
+		err = stringErr(fmt.Sprint(r))
```

Comment:
yeah, it would, it just felt silly to alloc the wrapper just to then have raven call `.Error()` on it in the next instruction, but we're crashing anyway, so the alloc isn't important.